### ISSUE TITLE
new rule: no-multiple-assertions-wait-for

### ISSUE THREAD

**Issue by renatoagds** [testing-library/eslint-plugin-testing-library#133]
Following the proposed in Kent at his post (https://kentcdodds.com/blog/common-mistakes-with-react-testing-library). I would like to suggest a new rule that detects multiple expect inside a wait-for:

```javascript
❌ // wrong
await waitFor(() => {
  expect(a).toEqual('a')
  expect(b).toEqual('b')
})
```

```javascript
✅ // correct
await waitFor(() => expect(a).toEqual('a'))
expect(b).toEqual('b')
```

**Comment by Belco90** [comment]
Hey Renato! Thanks for your proposal, it sounds good. Would like to give it a try? PRs are welcome!

**Comment by Belco90** [comment]
What name do you prefer for this one?
- `no-multiple-expect-wait-for`
- `single-wait-for-assert`
- something else

**Comment by nickmccurdy** [comment]
Maybe `no-multiple-assertions-wait-for`?

**Comment by gndelia** [comment]
should we also cover the deprecated methods? like wait and waitForElement?

**Comment by Belco90** [comment]
I think so. It should be easy after implemented for `waitFor` right?

**Comment by timdeschryver** [comment]
While being easy, is it needed? 
The linter will fix the deprecated methods to `waitFor`, and then this rule kicks in.

Just thinking out loud here 😅

**Comment by renatoagds** [comment]
@Belco90 @gndelia Hey guys! I have been working on this one since few days ago. Sorry for the delay, but I'm calling it `no-multiple-assertions-wait-for` in my branch: https://github.com/renatoagds/eslint-plugin-testing-library/tree/no-multiple-expect-wait-for

**Comment by gndelia** [comment]
oh nono, that's great! Let's continue over your branch. We did not know you were at it. 🚀 🚀 

you might want to join the debate here 😄

**Comment by renatoagds** [comment]
@gndelia Great! Will continue work on this.

> you might want to join the debate here

Yes! Sorry for the delay, I read the discussion from #165 and up to date about the directions.

> While being easy, is it needed?

@timdeschryver I have mixed feelings about. Talking about myself, the project that I work, we're in top of migration for RTL v10, since we have a lot of tests and even with lint fix we have a few points to double check, so we're thinking about use TL eslint to ensure good practices without wait for migration, in this case, could be good to support deprecated methods, but, since it's a opt-in rule, I could only enable it in the final of the migration.

I would suggest to start with `waitFor` and receive feedback if people are trying to use it with deprecated, and implement if necessary.

**Comment by Belco90** [comment]
@renatoagds sorry, didn't know you were still working on it! I'm happy with your suggestion to start only with `waitFor` and see what feedback we get. It's a good idea also to rename the rule to `no-multiple-assertions-wait-for`. Thanks!
